adapters/eea-direct.js

```diff
@@ -0,0 +1,265 @@
+    },
+    (done) => {
+      // try to combine locations with coordinates
+      // by matching stationId or a spatial join.
+      // then return successful combos
+      request.get({url: stationsLink}, (err, res, locations) => {
+        if (err) {
+          return done(null, []);
+        }
+        locations = JSON.parse(locations);
+        map(coordinates, (crds, cb) => {
+          if (err) {
+            return cb(null, crds);
+          }
+          let geocodedProps = locations.find((location) => {
+            // an object with city, region, bounds, and stationId
+            // or just a stationId if reverse geocoding was unsuccessful
+            if (location.stationId) {
+              return location.stationId === crds.stationId;
+            } else {
+              return location === crds.stationId;
+            }
+          });
+          // if new station, denoted by no matching id,
+          // try to spatial join
+          if (!(geocodedProps)) {
+            geocodedProps = locations.find((location) => {
+              if (location.stationId) {
+                const bbox = bboxPolygon(location.bounds);
+                const stationPoint = point(
+                  crds.longitude,
+                  crds.latitude
+                );
+                return inside(stationPoint, bbox);
+              }
+            });
+          }
+          if (geocodedProps) {
+            geocodedProps = typeof geocodedProps === 'string' ? crds : Object.assign(geocodedProps, crds);
+            cb(null, geocodedProps);
+          }
+        }, (err, updatedCoordinates) => {
+          if (err) {
+            return done(null, coordinates);
+          }
+          done(null, updatedCoordinates);
+        });
+      });
+    }
+  ], (err, mappedData) => {
+    if (err) {
+      return cb(err, []);
+    }
+    let measurements = mappedData[0];
+    const coordinates = mappedData[1];
+    // map coordinates and location names to measurements
+    map(measurements, (measurement, done) => {
+      let station = matchStation(coordinates, measurement['coordinates']);
+      measurement['location'] = (station.location ? station.location : 'unused');
+      measurement['city'] = (station.city ? station.city : 'unused');
+      done(null, measurement);
+    }, (err, finalMeasurements) => {
+      if (err) {
+        return cb(null, {name: 'unused', measurements: measurements});
+      }
+      cb(null, {name: 'unused', measurements: finalMeasurements});
+    });
+  });
+};
+
+const matchStation = (coordinates, measurementCoords) => {
+  let station;
+  if (coordinates.length > 0) {
+    station = coordinates.find((crds) => {
+      return crds.coordinates === measurementCoords;
+    });
+    return station;
+  }
+};
+
+const makeCoordinates = (coordinatesList, stationId) => {
+  return coordinatesList.find((coordinates) => {
+    return coordinates.stationId === stationId;
+  }).coordinates;
+};
+
+const makeDate = (date, timeZone) => {
```

Comment:
This seems overly complicated. We're gettign a timezone offset, and then tying it to a country's timezone, presumed to be in that area. But country timezones change at different times throughout the year. I think we should be directly using the data that they're given us, not proxying to some other country. 

Take a look at what the current EEA adapter does, I think the same should work here? https://github.com/openaq/openaq-fetch/blob/develop/adapters/eea.js#L94-L101

Reply:
good catch @olafveerman - since we have the lat/lon and moment like timezone strings, could we use something like [this](https://github.com/candu/efele-tz-world-geojson). We could do a spatial select (Getting polygon attrib) and use that as the tz string we pass to moment.